src/workerd/io/actor-sqlite.c++

```diff
@@ -464,6 +469,21 @@ ActorCacheInterface::DeleteAllResults ActorSqlite::deleteAll(WriteOptions option
   }
 
   uint count = kv.deleteAll();
+
+  // TODO(correctness): Since workerd doesn't have a separate durability step, in the unlikely
+  // event of a failure here, between deleteAll() and setAlarm(), we could theoretically lose the
+  // current alarm state when running under workerd.  Not sure if there's a practical way to avoid
+  // this.
+
+  // Reinitialize metadata wrapper, to allow on-demand recreation of metadata table.
+  metadata = SqliteMetadata(*db);
```

Comment:
Hmm, overwriting using an assignment like this is sort of surprising -- I wouldn't have expected `SqliteMetadata` to be assignable. It's also a little inefficient in that it's going to perform a database query to check for the existence of the metadata table, which in this case we know is definitely not there.

I think instead SqliteMetadata should probably register itself as a reset listener on the SqliteDatabase, and in its `beforeReset()` callback it should de-initialize itself.

Reply:
Implemented in bee6326be .